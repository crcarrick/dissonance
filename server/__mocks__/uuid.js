export const v1 = jest.fn();
export const v4 = jest.fn();
