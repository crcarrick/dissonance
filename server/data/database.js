import { Model, Sequelize } from 'sequelize';

import { channel } from './channel/channel.model';
import { message } from './message/message.model';
import { server } from './server/server.model';
import { user } from './user/user.model';

export const connectDatabase = () => {
  const sequelize = new Sequelize(
    process.env.DATABASE_NAME,
    process.env.DATABASE_USERNAME,
    process.env.DATABASE_PASSWORD,
    {
      host: 'localhost',
      dialect: 'postgres',
      logging: false,
    }
  );

  const Channel = channel({ sequelize });
  const Message = message({ sequelize });
  const Server = server({ sequelize });
  const User = user({ sequelize });

  class UserServer extends Model {}

  UserServer.init({}, { sequelize, tableName: 'users_servers' });

  const models = { Channel, Message, Server, User, UserServer };

  Object.keys(models).forEach((key) => {
    if ('associate' in models[key]) {
      models[key].associate(models);
    }
  });

  return { models, sequelize };
};
