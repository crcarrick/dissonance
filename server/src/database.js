export const createDbClient = (knex) =>
  knex(require('./../knexfile')[process.env.NODE_ENV]);
