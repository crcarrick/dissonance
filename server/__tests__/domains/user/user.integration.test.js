import { gql } from 'apollo-server';
import aws from 'aws-sdk';
import { v4 as uuidv4 } from 'uuid';

import { createDbClient } from '@dissonance/database';

import {
  createTestClient,
  generateMockData,
  seedDatabase,
} from '@dissonance/test-utils';

const knex = jest.requireActual('knex');

describe('User Integration', () => {
  const { users } = generateMockData();

  let dbClient;
  beforeAll(() => {
    dbClient = createDbClient(
      knex,
      `dissonance_test_${process.env.JEST_WORKER_ID}`
    );
  });

  afterAll((done) => {
    dbClient.destroy().then(() => done());
  });

  let gqlClient;
  beforeEach(async (done) => {
    await seedDatabase(dbClient);

    gqlClient = createTestClient({
      context: {
        user: users[0],
      },
      dbClient,
    });

    done();
  });

  test('creates a signed url for avatar upload when provided an image', async () => {
    aws.s3Mock.getSignedUrlPromise.mockReturnValueOnce('https://www.test.com');
    uuidv4.mockReturnValueOnce('1234');

    const response = await gqlClient.mutate({
      mutation: gql`
        mutation CreateUserAvatarSignedUrl($input: CreateUserSignedUrlInput!) {
          createUserAvatarSignedUrl(input: $input) {
            signedUrl
            url
          }
        }
      `,
      variables: {
        input: {
          fileName: 'test.png',
        },
      },
    });

    expect(response.data.createUserAvatarSignedUrl).toMatchSnapshot();
  });

  test('does not create a signed url for avatar upload when provided a non-image', async () => {
    const response = await gqlClient.mutate({
      mutation: gql`
        mutation CreateUserAvatarSignedUrl($input: CreateUserSignedUrlInput!) {
          createUserAvatarSignedUrl(input: $input) {
            signedUrl
            url
          }
        }
      `,
      variables: {
        input: {
          fileName: 'test.pdf',
        },
      },
    });

    expect(response.errors).toMatchSnapshot();
  });
});
