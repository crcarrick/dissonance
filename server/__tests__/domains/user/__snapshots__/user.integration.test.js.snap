// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Integration creates a signed url for avatar upload when provided an image 1`] = `
Object {
  "signedUrl": "https://www.test.com/",
  "url": "https://dissonance.s3.amazonaws.com/1234",
}
`;

exports[`User Integration does not create a signed url for avatar upload when provided a non-image 1`] = `
Array [
  [GraphQLError: File is not an image type],
]
`;
